### PR TITLE
Template configuration state, and rotation fix

### PR DESCRIPTION
**Pull request by dg0yt**
This is a broader revision of the setup and configuration of templates. It includes an adaptation of the OgrTemplate rotation fix submitted by @pkturner in GH-1659.

### CODE REVIEW

**Comment by dg0yt** [comment]
@pkturner I did following adaptations:
- Add basic testing of TemplateTrack and OgrTemplate, based on your test files and code.
- Consolidate the rotation test, have the unloaded case, the loaded cases, and the loaded-then-unloaded case for TemplateTrack and OgrTemplate all in one file (with the option to extent this to other templates classes.)
- The finish-configuration hook returns `bool`, as do most of our related virtual functions.

File: src/templates/template.h [hunk 1]
```diff
@@ -646,6 +646,22 @@ Q_OBJECT
...
 	virtual bool loadTypeSpecificTemplateConfiguration(QXmlStreamReader& xml);
 	
+	/**
+	 * Hook which is called at the end of template configuration.
+	 * 
+	 * Derived classes may override this function if they need to do any post-
+	 * processing on the configuration. Unlike loadTypeSpecificTemplateConfiguration(),
+	 * this function is always called, even when there is no type-specific data.
+	 * 
+	 * The implementation must not do expensive calculations because is called
```
Comment (dg0yt) [inline src/templates/template.h:658]: "it is called"

File: src/templates/template.h [hunk 2]
```diff
@@ -646,6 +646,22 @@ Q_OBJECT
 	 */
 	virtual bool loadTypeSpecificTemplateConfiguration(QXmlStreamReader& xml);
 	
+	/**
+	 * Hook which is called at the end of template configuration.
```
Comment (dg0yt) [inline src/templates/template.h:652]: "template configuration loading."

**Comment by dg0yt** [comment]
> Add basic testing of TemplateTrack and OgrTemplate, based on your test files and code.

.. and in the CI build without GDAL, this immediately turns out a bug.

**Comment by pkturner** [comment]
It looks good. I'm glad #1659 was useful.

**Pull request merged**